### ISSUE TITLE
Can't push schema

### ISSUE THREAD

**Issue by Bulletninja** [aws-amplify/amplify-cli#4903]
My use with AWS amplify has been a love-hate relationship. Simple scenarios are very easy, but then it breaks a lot. Right now whatever change i do to my `schema.graphql` file, i get the same error, i've been restoring it each time to try and figure out how to solve this, but haven't found a way. The error is:

`Updates are not allowed for property - UsernameAttributes`

just from pushing the api.

Any ideas how to fix it?

**Comment by nikhname** [comment]
Hi @Bulletninja, could you provide a snippet of your schema that contains `UsernameAttributes`?

**Comment by Bulletninja** [comment]
There is not "UsernameAttributes" in my schema, at least not in graphql, it seems to me it's hidden somewhere else. Even if i search my codebase, i was only to find an instance where i was call "withAuthenticator" like this:
`withAuthenticator(AuthProvider, { usernameAttributes: 'email', signupConfig});`

But that's not the schema, and the error persists if i remove that.

**Comment by Bulletninja** [comment]
Out of frustration i left aws amplify since this issue (basically, i have a love-hate relationship with aws amplify, because it solves a lot of problems, but i keep jumping from issue to issue, i'm always "almost there" but not quite) temporarily. 
I did a quick check, and tried to push again. 
I got 
```
UPDATE_FAILED               authsuscribela0aee2191           AWS::CloudFormation::Stack Mon Aug 03 2020 22:37:11 GMT-0600 (Mexican Pacific Daylight Time) Parameters: [unauthRoleName, authRoleName] do not exist in the template
```
I'm not sure why, but this is one of those errors that keep popping randomly (seems to me) from time to time, this is one of those frustration sources. But i'm unsure why the error changed since i didn't touch the codebase since getting this error according to my repo :/
